### PR TITLE
Stop allowing configurable z and zo commands

### PR DESCRIPTION
**Pull request by ismay**
This implements what we discussed here: https://github.com/jethrokuan/z/issues/107#issuecomment-1085674474. By moving to a static command we can ship the command and its completions in the format that fisher expects. This isn't just useful for fisher, it also works better with fish. By shipping static commands fish can lazy load both the commands and their completions. This means we don't have to run setup for this in conf.d scripts. It also simplifies the code a little bit.

Users can still easily create an alias fn, which is the canonical fish way of renaming a command. That way, even if they want to alias the command they can still retain the lazy loading.

Also, I took a look with grep.app, and could find only two users who were using a custom z command: [see here](https://grep.app/search?q=%24Z_CMD&case=true&filter[lang][0]=fish). So this seems ok to change if we want to.

### CODE REVIEW

File: conf.d/z.fish [hunk 1]
```diff
@@ -14,50 +14,25 @@ if test ! -e "$Z_DATA"
...
 end
 
-if test -z "$Z_CMD"
-    set -U Z_CMD z
-end
-
-set -U ZO_CMD "$Z_CMD"o
-
-if test ! -z $Z_CMD
-    function $Z_CMD -d "jump around"
```
Comment (ismay) [inline conf.d/z.fish:24]: I think `jump around` and `open target dir` (line 30 of this file), were the descriptions that were used by fish for these functions. I kept the one that was in `__z.fish`, but if you want we can use these instead.

**Comment by ismay** [comment]
I've rebased on master. I think all that's left now is to regenerate the man pages. Do we have to do that manually?

File: man/man1/z.md [hunk 2]
```diff
@@ -60,9 +60,6 @@ For more details about frecency, see https://github.com/rupa/z.
 
 ## CONFIGURATION
 
-  * `set -U Z_CMD "j"`:
-    Change commands to `j` and `jo`
-
```
Comment (septatrix) [inline man/man1/z.md:65]: Maybe explain somewhere that aliases can be made (preferably) with `abbr` or `function --wraps`

Comment (ismay) [reply man/man1/z.md:65]: Yeah that sounds good. I don't really have the time anymore to work on this though.

**Pull request closed without merging**